File: pubsub/src/main/java/glitch/pubsub/object/json/Response.java
package glitch.pubsub.object.json;

import com.google.gson.annotations.JsonAdapter;
import glitch.pubsub.exceptions.PubSubException;
import glitch.pubsub.object.adapters.ExceptionAdapter;
import glitch.pubsub.object.adapters.MessageTypeAdapter;
import glitch.pubsub.object.enums.MessageType;
import lombok.Data;

@Data
public class Response {
    @JsonAdapter(MessageTypeAdapter.class)
    private final MessageType type;
    private final String nonce;
    @JsonAdapter(ExceptionAdapter.class)
    private final PubSubException error;
    private final ResponseData data;
}

File: pubsub/src/main/kotlin/glitch/pubsub/object/json/Response.kt
package glitch.pubsub.`object`.json

import com.google.gson.annotations.JsonAdapter
import glitch.pubsub.`object`.adapters.ExceptionAdapter
import glitch.pubsub.`object`.adapters.MessageTypeAdapter
import glitch.pubsub.`object`.enums.MessageType
import glitch.pubsub.exceptions.PubSubException

class Response(
        @JsonAdapter(MessageTypeAdapter::class)
        val type: MessageType,
        val nonce: String?,
        @JsonAdapter(ExceptionAdapter::class)
        val error: PubSubException?,
        val data: ResponseData?
) {
}
